### DISCUSSION TITLE
Modify the default File menu

### DISCUSSION THREAD

**Discussion by ilyagr** [tauri-apps/tauri#8853] [Q&A] [answered]
I'd like to start with the default menu on Mac OS, but then to add some items to the File menu (e.g. Save). Is this possible? Currently, I can either get my File menu by starting with a completely new menu, or my additions to the File menu are ignored.

Cc: https://github.com/tauri-apps/tauri/issues/2398

**Comment by FabianLars** [comment, marked as answer]
This is only possible in v2. The v1 menu api is too limited and afaik can't be changed to make it possible either.

In v2 you can get the current Menu via `window.menu()` and then you'll have access to all those apis here https://docs.rs/tauri/2.0.0-beta.2/tauri/menu/struct.Menu.html, including an `insert()` function.

**Reply by ilyagr** [reply]
Makes sense, thank you!

**Update:** According to the docs, there's a limitation on Mac OS for now: only Submenus can be added to menus. I haven't looked in enough detail to check whether items like File -> Save can be added, but clearly there are people interested in improving it.

**Reply by FabianLars** [reply]
"File" is the sub menu in this context.
